x/application/keeper/query_application_test.go

```diff
@@ -122,3 +124,44 @@ func TestApplicationQueryPaginated(t *testing.T) {
 		require.ErrorIs(t, err, status.Error(codes.InvalidArgument, "invalid request"))
 	})
 }
+
+func TestAllApplicationsQuery_WithDelegateeGatewayAddressConstraint(t *testing.T) {
+	keeper, ctx := keepertest.ApplicationKeeper(t)
+	gatewayAddr1 := sample.AccAddress()
+	appsWithDelegationAddr := []string{"1", "2"}
+	apps := createNApplications(keeper, ctx, 5, testAppModifierDelegateeAddr(gatewayAddr1, appsWithDelegationAddr))
+
+	requestBuilder := func(gatewayAddr string) *types.QueryAllApplicationsRequest {
+		return &types.QueryAllApplicationsRequest{
+			DelegateeGatewayAddress: gatewayAddr,
+		}
+	}
+
+	t.Run("QueryAppsWithDelegatee", func(t *testing.T) {
+		resp, err := keeper.AllApplications(ctx, requestBuilder(gatewayAddr1))
+		require.NoError(t, err)
+
+		var expectedApps []types.Application
+		for _, app := range apps {
+			if slices.Contains(appsWithDelegationAddr, app.Address) {
+				expectedApps = append(expectedApps, app)
+			}
+		}
+
+		require.ElementsMatch(t,
+			nullify.Fill(expectedApps),
+			nullify.Fill(resp.Applications),
+		)
+	})
+
+	t.Run("QueryAppsWithInvalidGatewayAddr", func(t *testing.T) {
+		addrInvalid := "invalid-address"
+		_, err := keeper.AllApplications(ctx, requestBuilder(addrInvalid))
+		expectedErr := status.Error(
+			codes.InvalidArgument,
+			types.ErrAppInvalidGatewayAddress.Error(),
+		)
+
+		require.ErrorContains(t, err, expectedErr.Error())
+	})
+}
```

Comment:
It's worth adding a test with an empty address and assert that all apps are returned.

Reply:
+1